templates/vue3/src/App.vue

```diff
@@ -0,0 +1,160 @@
+                    </li>
+                    <li>
+                        When you run <code>npm run dev</code> or <code>yarn dev</code>, the code in <code>assembly/main.ts</code> 
+                        gets deployed to the NEAR testnet. You can see how this happens by looking in <code>package.json</code> 
+                        at the <code>scripts</code> section to find the <code>dev</code> command.
+                    </li>
+                </ol>
+                <hr />
+                <p>
+                    To keep learning, check out <a target="_blank" rel="noreferrer" href="https://docs.near.org">the NEAR docs</a> or look through some <a target="_blank" rel="noreferrer" href="https://examples.near.org">example apps</a>.
+                </p>
+            </main>
+            <Notification v-show="notificationVisible" 
+                          ref="notification"
+                          :networkId="networkId"
+                          :msg="'called method: setGreeting'"
+                          :visible="false"
+                          />
+        </template>
+
+        <template v-else>
+            <main>
+                <h1>Welcome to NEAR!</h1>
+                <p>
+                    To make use of the NEAR blockchain, you need to sign in. The button
+                    below will sign you in using NEAR Wallet.
+                </p>
+                <p>
+                    By default, when your app runs in "development" mode, it connects
+                    to a test network ("testnet") wallet. This works just like the main
+                    network ("mainnet") wallet, but the NEAR Tokens on testnet aren't
+                    convertible to other currencies - they're just for testing!
+                </p>
+                <p>
+                    Go ahead and click the button below to try it out:
+                </p>
+                <p style="text-align:center; margin-top:2.5em">
+                    <button v-on:click="login">Sign in</button>
+                </p>
+            </main>
+        </template>
+    </div>
+</template>
+
+<script>
+
+import getConfig from './config'
+const nearConfig = getConfig(process.env.NODE_ENV || 'development')
+
+console.log(`networkId:${nearConfig.networkId} CONTRACT_NAME:${nearConfig.contractName}`);
+
+import { login, logout, onSubmit } from './utils'  
+import './global.css'
+
+import Notification from './components/Notification.vue'
+
+    export default {
+        created() {
+            if (this.signedIn) {
+                this.retrieveSavedGreeting()
+            }
+        },
+        name: 'App',
+        components: {
+            Notification
+        },
+        data: function () {
+            return {
+                savedGreeting: "Hello stranger",
+                newGreeting: "Hello stranger",
```

Comment:
are these defaults? seems strange to default them to “Hello stranger” when the default the contract will return is just “Hello”. I’d rather they defaulted to empty strings or null

Reply:
I'll change it to "Hello"

Reply:
After someone has changed the greeting and they refresh the page, will setting this to "Hello" cause "Hello" to flash briefly on the screen before the updated greeting loads? I'd rather have an empty string or `...` show up. Seeing "Hello" briefly seems like it could cause confusion.